src/control/server/storage/bdev.go

```diff
@@ -24,6 +31,61 @@ import (
 // BdevPciAddrSep defines the separator used between PCI addresses in string lists.
 const BdevPciAddrSep = " "
 
+// BioState represents the health state of BIO device on an NVMe device (a blobstore on a NVMe
+// namespace).
+type BioState uint32
+
+// BioState constant definitions to represent expected bitset flag combinations.
+const (
+	BioStateNew      BioState = C.NVME_DEV_FL_PLUGGED
+	BioStateNormal   BioState = C.NVME_DEV_FL_PLUGGED | C.NVME_DEV_FL_INUSE
+	BioStateFaulty   BioState = C.NVME_DEV_FL_PLUGGED | C.NVME_DEV_FL_INUSE | C.NVME_DEV_FL_FAULTY
+	BioStateIdentify BioState = C.NVME_DEV_FL_PLUGGED | C.NVME_DEV_FL_INUSE | C.NVME_DEV_FL_IDENTIFY
+)
+
+// IsNew returns true if SSD is not in use by DAOS.
+func (bs BioState) IsNew() bool {
+	return bs&C.NVME_DEV_FL_INUSE == 0
+}
+
+// IsNormal returns true if SSD is in a normal, non-faulty state.
+func (bs BioState) IsNormal() bool {
+	return (bs&C.NVME_DEV_FL_PLUGGED != 0 && bs&C.NVME_DEV_FL_INUSE != 0 &&
+		bs&C.NVME_DEV_FL_FAULTY == 0)
+}
+
+// IsFaulty returns true if SSD is in a faulty state.
+func (bs BioState) IsFaulty() bool {
+	return bs&C.NVME_DEV_FL_FAULTY != 0
+}
+
+func (bs BioState) String() string {
+	switch {
+	case bs&C.NVME_DEV_FL_PLUGGED == 0:
+		return "UNPLUGGED"
+	case bs&C.NVME_DEV_FL_INUSE == 0:
+		return "NEW"
+	case bs&C.NVME_DEV_FL_FAULTY != 0:
+		return "EVICTED"
+	case bs&C.NVME_DEV_FL_IDENTIFY != 0:
+		return "IDENTIFY"
```

Comment:
Not sure I understand this IDENTIFY state or why it is added to the BIO state. The identify state for the LED only lasts for 1 min and then returns to off. So not sure we would want to display that as a state?

Reply:
consolidated the various states into one bitset, we don't have to display it as a state but I don't see how it hurts and it seems consistent